#!/usr/bin/env python

# set up parameters that we care about
PACKAGE = 'jsk_pcl_ros'

from dynamic_reconfigure.parameter_generator_catkin import *;

from math import pi

gen = ParameterGenerator ()

gen.add("min_radius", double_t, 0, "minimum radius", 0.05, 0.01, 1.0)
gen.add("max_radius", double_t, 0, "maximum radius", 0.2, 0.01, 1.0)
gen.add("filter_distance", double_t, 0, "filter distance by line", 0.2, 0.01, 1.0)
gen.add("outlier_threshold", double_t, 0, "outlier_threshold", 0.01, 0.001, 0.1)
gen.add("max_iteration", int_t, 0, "max iteration", 100, 1, 10000)
gen.add("eps_angle", double_t, 0, "eps angle", 0.1, 0.0, pi)
gen.add("min_probability", double_t, 0, "minimum probability", 0.8, 0.0, 1.0)
gen.add("cylinder_fitting_trial", int_t, 0, "maximum trial number to fit cylinder model", 3, 1, 10)
gen.add("min_inliers", int_t, 0, "minimum number of inliers", 10, 0, 10000)
gen.add("eps_2d_angle", double_t, 0, "eps 2D angle", 0.1, 0, pi)
exit (gen.generate (PACKAGE, "jsk_pcl_ros", "HintedStickFinder"))
